src/firefly/java/edu/caltech/ipac/firefly/server/visualize/WebPlotFactory.java

```diff
@@ -120,10 +124,24 @@ private static void writeAnyModifiedFiles(ImagePlotCreator.PlotInfo[] pInfo) {
         for(var pi : pInfo) {
             PlotState state= pi.state();
             for (Map.Entry<Band, ModFileWriter> entry : pi.fileWriterMap().entrySet()) {
+                Band band = entry.getKey();
                 ModFileWriter mfw = entry.getValue();
                 if (mfw != null) {
                     pi.state().setImageIdx(0, entry.getKey());
                     mfw.writeFile(state);
+                    try {
+                        // re-create FitsRead arrays after files are written to disk to ensure correct SPOT_HS headers
+                        File workingFile = PlotStateUtil.getWorkingFitsFile(state, band);
+                        if (workingFile != null && workingFile.exists()) {
+                            Fits fits = new Fits(workingFile);
+                            FitsRead[] fitsReadArray = FitsReadFactory.createFitsReadArray(fits);
+                            pi.fitsReadGroup().setFitsRead(band, fitsReadArray[0]);
+                        }
```

Comment:
The logic looks good. However, I think you could modify `mfw.writeFile` to take more parameters and just to it in that function to make the overall code more readable.

so in ModFileWriter, change the signature to
```java
public void writeFile(PlotState state, ActiveFitsReadGroup frGroup) 
```
and put the re-create FitsRead arrays part in there